### PR TITLE
Accept objects as className in styled components

### PR DESCRIPTION
**Pull request by jgroszko**
**What**: Accept objects as className in styled components

**Why**: Matches react behavior per issue #1495 

**How**: Add a special case for objects that don't already have a 'styles' property

**Checklist**:

- [ N/A ] Documentation
- [x] Tests
- [x] Code complete
- [x] Changeset

### CODE REVIEW

File: packages/styled-base/src/index.js [hunk 1]
```diff
@@ -106,6 +106,17 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
                 props.className
               )
             }
+            if (
```
Comment (Andarist) [inline packages/styled-base/src/index.js]: IMHO this should just be else if to the preceding if statement, like this:
```js
if (typeof props.className === 'string') {
  // ...
} else if (props.className != null) {
  className += props.className
}
```

Comment (jgroszko) [reply packages/styled-base/src/index.js]: Ahhh yes that does make sense

File: packages/styled-base/src/index.js [hunk 2]
```diff
@@ -105,7 +105,17 @@ let createStyled: CreateStyled = (tag: any, options?: StyledOptions) => {
                 classInterpolations,
                 props.className
               )
+            } else if (
```
Comment (Andarist) [inline packages/styled-base/src/index.js]: Maybe that wasn't clear - but in my last comment I've also included alternative solution to the "problem", the condition of this else if should be just `props.className != null` and the body of this shouldn't call `getRegisteredStyles`, like this:
```js
if (typeof props.className === 'string') {
  // ...
} else if (props.className != null) {
  className += `${props.className} `
}
```

Comment (jgroszko) [reply packages/styled-base/src/index.js]: Ahhh okay, I see what you mean

**Comment by Andarist** [comment]
Seems like you have uncovered a mistake in tests. This line:
https://github.com/emotion-js/emotion/blob/a07f8a73d9b2a567dbb4e68c58d2c304ae9ea833/packages/styled/__tests__/styled.js#L395
should use `css` prop and not `className`.

File: .changeset/afraid-dolls-speak.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@emotion/styled-base': minor
```
Comment (Andarist) [inline .changeset/afraid-dolls-speak.md]:
```suggestion
'@emotion/styled-base': patch
```

File: .changeset/afraid-dolls-speak.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+'@emotion/styled': minor
```
Comment (Andarist) [inline .changeset/afraid-dolls-speak.md]:
```suggestion
'@emotion/styled': patch
```

File: .changeset/afraid-dolls-speak.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Accept objects as className in styled components
```
Comment (Andarist) [inline .changeset/afraid-dolls-speak.md]:
```suggestion
Accept objects as `className` in styled components (they are stringified) to match React behavior
```

**Comment by Andarist** [comment]
Thanks!

**Pull request merged**